functions/module_metadata/assert.pp

```diff
@@ -0,0 +1,69 @@
+# Fails a compile if the client system is not compatible with the module's
+# `metadata.json`
+#
+# @param module_name
+#   The name of the module that should be checked
+#
+# @param options
+#   @see $simplib::module_metadata::blacklist::options
+#
+# @return [None]
+#
+function simplib::module_metadata::assert (
+  String[1] $module_name,
+  Optional[Struct[{
+    enable => Optional[Boolean],
+    blacklist => Optional[Array[Variant[String[1], Hash[String[1], Variant[String[1], Array[String[1]]]]]]],
```

Comment:
Con't you want the blacklist Array size to be at least of length 1.  Otherwise the code below should check if the blacklist Array is empty.

Reply:
Nah, an empty Array just returns false.